tests/unit/objectives/test_utils.py

```diff
@@ -49,3 +56,51 @@ def test_mk_multi_observer() -> None:
     npt.assert_array_equal(ys["foo"].observations, x_ + 1)
     npt.assert_array_equal(ys["bar"].query_points, x_)
     npt.assert_array_equal(ys["bar"].observations, x_ - 1)
+
+
+@pytest.mark.parametrize(
+    "input_objective, keys",
+    [
+        (lambda x: x, ["baz"]),
+        (lambda x: Dataset(x, x), ["baz"]),
+        (mk_multi_observer(foo=lambda x: x + 1, bar=lambda x: x - 1), ["foo", "bar"]),
+    ],
+)
+@pytest.mark.parametrize("batch_size", [1, 2, 3])
+@pytest.mark.parametrize("num_query_points_per_batch", [1, 2])
+def test_mk_batch_observer(
+    input_objective: Union[Callable[[TensorType], TensorType], Observer],
+    keys: Sequence[Tag],
+    batch_size: int,
+    num_query_points_per_batch: int,
+) -> None:
+    x_ = tf.reshape(
+        tf.constant(range(batch_size * num_query_points_per_batch), tf.float64),
+        (num_query_points_per_batch, batch_size, 1),
+    )
+    ys = mk_batch_observer(input_objective, "baz")(x_)
+
+    assert isinstance(ys, dict)
+
+    # Check keys.
+    exp_keys: Set[Union[Tag, LocalizedTag]] = set()
+    for key in keys:
+        exp_keys.update({LocalizedTag(key, i) for i in range(batch_size)})
+        exp_keys.add(key)
+    assert ys.keys() == exp_keys
+
+    # Check datasets.
+    for key in keys:
+        # Different observers (in parameterize above) return different observation values.
```

Comment:
(might be cleaner to move this to the parametrize? maybe replace "keys" with a mapping of keys to lambda?))